### PR TITLE
Convert to TypeScript

### LINKED ISSUE #3: Convert to TypeScript
[Issuehunt badges](https://issuehunt.io/r/egoist/vue-compile/issues/3)

💪🏻

---

<details>
<summary>
<b>IssueHunt Summary</b>
</summary>

### Sponsors (Total: $200.00)

- [issuehunt issuehunt](https://issuehunt.io/u/issuehunt) ($200.00)

### Tips

- Checkout the [Issuehunt explorer](https://issuehunt.io/r/egoist/vue-compile/) to discover more funded issues.
- Need some help from other developers? [Add your repositories](https://issuehunt.io/r/new) on IssueHunt to raise funds.
</details>

### PR DESCRIPTION
**Pull request by hawx1993**
Convert to TypeScript

Closes #3

### CODE REVIEW

File: .eslintrc [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+      "jsx": false
+    },
+    "project": "./tsconfig.json"
+  },
+  "extends": [
+    "eslint:recommended",
+    "plugin:@typescript-eslint/eslint-recommended",
+    "plugin:@typescript-eslint/recommended"
+  ]
+}
```
Comment (egoist) [inline .eslintrc]: this project uses XO.

File: bin/cli.ts [hunk 2]
```diff
@@ -1,5 +1,9 @@
 #!/usr/bin/env node
-const chalk = require('chalk')
+import chalk from 'chalk';
+import cac from 'cac';
+import pkg from '../package.json';
+import VueCompile from '../lib';
```
Comment (egoist) [inline bin/cli.ts]: we want to import vue-compile lazily

File: package.json [hunk 3]
```diff
@@ -38,9 +40,16 @@
     "postcss-load-config": "^2.0.0",
     "resolve": "^1.8.1",
     "stringify-attributes": "^1.0.0",
+    "typescript": "^3.5.1",
```
Comment (egoist) [inline package.json]: should be `devDependencies`

**Comment by egoist** [comment]
please update `package.json` accordingly. (we need to publish compiled code on npm)

File: .eslintrc (hunk 1 above)
Comment (hawx1993) [reply .eslintrc]: done

File: bin/cli.ts (hunk 2 above)
Comment (hawx1993) [reply bin/cli.ts]: done

File: package.json (hunk 3 above)
Comment (hawx1993) [reply package.json]: done

File: .eslintrc (hunk 1 above)
Comment (egoist) [reply .eslintrc]: that's not what I mean, see https://github.com/egoist/vue-compile/blob/d5c00129d0d6043a1194c7ebd2353dbdb21c2d77/package.json#L51

Comment (hawx1993) [reply .eslintrc]: sorry, still can't work. I set: 
```json
"xo": {
    "extends": "xo-typescript",
    "extensions": [
      "ts"
    ],
    "rules": {
      "unicorn/filename-case": "off"
    }
  },
```
but report: `Cannot find module 'eslint-plugin-@typescript-eslint'`

**Pull request closed without merging**